pkg/collector/corechecks/snmp/internal/profile/profile_resolver.go

```diff
@@ -0,0 +1,137 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package profile
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-agent/pkg/networkdevice/profile/profiledefinition"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/internal/common"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/internal/configvalidation"
+)
+
+func resolveProfiles(userProfiles, defaultProfiles ProfileConfigMap) (ProfileConfigMap, error) {
+	defaultExpandedProfiles, err := loadResolveProfiles(defaultProfiles, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load profiles: %s", err)
+	}
+	userExpandedProfiles, err := loadResolveProfiles(userProfiles, defaultProfiles)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load profiles: %s", err)
+	}
+	profiles := ProfileConfigMap{}
+	for key, val := range defaultExpandedProfiles {
+		if strings.HasPrefix(key, "_") {
```

Comment:
> if the users choose to monitor their devices using an abstract profile by specifying it in the instance config

This is the behaviour prior this PR and we likely want to keep it for now since the intent of the abstract profiles is to not be used directly.